robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -1,191 +1,545 @@
+                $('#goal-DMS-sec-input.lat.goal-' + current).attr("value", secs)
+            } else {
+                throw "You didn't enter anything for lat"
+            }
+            if (long_format == 0) {
+                //decimal-degrees mode
+                long = parseFloat($('#goal-DD-dec-deg-input.long.goal-' + current).val())
+
+                if (isNaN(long)) throw "Bad input long DD"
+
+                $('#goal-DD-dec-deg-input.long.goal-' + current).attr("value", long)
+            } else if (long_format == 1) {
+                //degrees-decimal-minutes mode
+                let dec = parseFloat($('#goal-DDM-deg-input.long.goal-' + current).val())
+                let min = parseFloat($('#goal-DDM-dec-min-input.long.goal-' + current).val())
+
+                if (isNaN(dec) || isNaN(min)) throw "Bad input long DDM"
+
+                long = dec + (min / 60)
+
+                $('#goal-DDM-deg-input.long.goal-' + current).attr("value", dec)
+                $('#goal-DDM-dec-min-input.long.goal-' + current).attr("value", min)
+
+            } else if (long_format == 2) {
+                //degrees-minutes-seconds mode
+                let dec = parseFloat($('#goal-DMS-deg-input.long.goal-' + current).val())
+                let mins = parseFloat($('#goal-DMS-min-input.long.goal-' + current).val())
+                let secs = parseFloat($('#goal-DMS-sec-input.long.goal-' + current).val())
+
+                if (isNaN(dec) || isNaN(mins) || isNaN(secs)) throw "Bad input long DMS"
+
+                long = dec + (mins / 60 + secs / 3600)
+
+                $('#goal-DMS-deg-input.long.goal-' + current).attr("value", dec)
+                $('#goal-DMS-min-input.long.goal-' + current).attr("value", mins)
+                $('#goal-DMS-sec-input.long.goal-' + current).attr("value", secs)
+            } else {
+                throw "You didn't enter anything for long"
+            }
+
+            let goalElem = $('#goal-name.goal-' + current)
+            let goalName = goalElem.val()
+            if (goalName.length != 0) goalElem.attr('value', goalName)
+            else goalElem.attr('value', 'Goal-' + current)
+
+            $('#goal-name-fieldset.goal-' + current).prop('disabled', true)
+            $('#goal-lat-fieldset.goal-' + current).prop('disabled', true)
+            $('#goal-long-fieldset.goal-' + current).prop('disabled', true)
+
+
+            /*----------------------------------\
+            this section implements ROS goal node
+            \----------------------------------*/
+            let goalData = new ROSLIB.Message({
+                name: goalName,
+                long: long,
+                lat: lat
+            })
+
+            create_goal_publisher.publish(goalData)
+
+        } catch (e) {
+            $('#goal-confirm-btn.goal-' + current).prop('disabled', false)
+            $('#goal-change-btn.goal-' + current).prop('disabled', true)
+
+            rosLog('ROSWARN', e)
```

Comment:
this should be `rosWarn`, `rosLog` is just the helper function

Reply:
Fixed